ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -0,0 +1,422 @@
+            clearAvus(targetApp);
+            return;
+        }
+
+        AvuList avuList = ontologyUtil.convertHierarchiesToAvus(hierarchy);
+        serviceFacade.addAVUsToApp(targetApp, avuList, new AsyncCallback<List<Avu>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<Avu> result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.appClassified(targetApp.getName(), hierarchy.getLabel())));
+            }
+        });
+    }
+
+    void clearAvus(final App targetApp) {
+        AvuList avuList = avuFactory.getAvuList().as();
+        serviceFacade.setAppAVUs(targetApp, avuList, new AsyncCallback<List<Avu>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<Avu> result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.appAvusCleared(targetApp)));
+            }
+        });
+    }
+
+    @Override
+    public void onViewOntologyVersion(ViewOntologyVersionEvent event) {
+        getOntologies();
+    }
+
+    void getOntologies() {
+        serviceFacade.getOntologies(new AsyncCallback<List<Ontology>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<Ontology> result) {
+                Collections.reverse(result);
+                view.showOntologyVersions(result);
+            }
+        });
+    }
+
+    @Override
+    public void onSelectOntologyVersion(SelectOntologyVersionEvent event) {
+        treeStore.clear();
+        iriToHierarchyMap.clear();
+        serviceFacade.getOntologyHierarchies(event.getSelectedOntology().getVersion(), new AsyncCallback<List<OntologyHierarchy>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(List<OntologyHierarchy> result) {
+                if (result.size() == 0) {
+                    view.showEmptyTreePanel();
+                } else {
+                    addHierarchies(null, result);
+                    view.showTreePanel();
+                }
+            }
+        });
+
+    }
+
+    void addHierarchies(OntologyHierarchy parent, List<OntologyHierarchy> children) {
+        if ((children == null)
+            || children.isEmpty()) {
+            return;
+        }
+        if (parent == null) {
+            ontologyUtil.addUnclassifiedChild(children);
+            treeStore.add(children);
+
+        } else {
+            treeStore.add(parent, children);
+        }
+
+        helperMap(children);
```

Comment:
This helperMap, `iriToHierarchyMap`, is needed mostly because if the admin wants to know what hierarchies an App is already tagged with, the metadata saved only includes the IRIs, not the paths.  Since an IRI can exist in a hierarchy multiple times, there's a need for a mapping from IRI to all the different hierarchies/paths.